RelayDotNet/Relay.cs

```diff
@@ -1946,5 +1949,186 @@ public async Task<string[]> SetUserProfile(IRelayWorkflow relayWorkflow, string
                 }
             );
         }
+
+        string serverHostname = "all-main-pro-ibot.nocell.io";
+        string version = "relay-sdk-dotnet/2.0.0";
+        string auth_hostname = "auth.relaygo.com";
+
+        private async Task<string> UpdateAccessToken(string refreshToken, string clientId) {
+            // Create a uri object with the following url
+            string url = $"https://{auth_hostname}/oauth2/token";
+            var grantUrl = new Uri(@url);
+
+            // Create a payload to be sent with the request
+            var grantPayload = new Dictionary<string, string>() {
+                ["grant_type"] = "refresh_token",
+                ["refresh_token"] = refreshToken,
+                ["client_id"] = clientId
+            };
+
+            // Create an http client that will post the request
+            HttpClient httpClient = new HttpClient();
+
+            // Add headers and the encoded content to the URL
+            httpClient.DefaultRequestHeaders.Accept.Add(
+                new MediaTypeWithQualityHeaderValue("application/json"));
+            httpClient.DefaultRequestHeaders.Add("User-Agent", version);
+            var encodedContent = new FormUrlEncodedContent(grantPayload);
+
+            // Post the request and retrieve the response
+            var grantResponse = await httpClient.PostAsync(grantUrl, encodedContent);
+
+            // If the resonse is not successful, throw an exception back to the client
+            if(grantResponse.StatusCode != System.Net.HttpStatusCode.OK) {
+                throw new Exception($"Unable to get access token: {grantResponse.StatusCode}");
+            }
+
+            // Create a json file with the grantResponse to make a dictionary, set the access 
```

Comment:
A nitpick: I don't think this is an actual file on disk but an in-memory data structure.